api/reqs/tests/model_tests.py

```diff
@@ -33,6 +33,19 @@ def test_original_url():
     assert policy.original_url == 'http://example.com/uploaded'
 
 
+@pytest.mark.parametrize('phase, has_document', (
+    (models.WorkflowPhases.edit, False),
+    (models.WorkflowPhases.cleanup, False),
+    (models.WorkflowPhases.failed, False),
+    (models.WorkflowPhases.no_doc, False),
+    (models.WorkflowPhases.published, True),
+    (models.WorkflowPhases.review, False),
+))
+def test_has_document(phase, has_document):
+    policy = models.Policy(workflow_phase=phase.name)
+    assert policy.has_document is has_document
```

Comment:
Hmm, so I am actually a bit confused by the name `has_document`. It definitely makes sense in the context of the viewer UI, but it doesn't necessarily make sense to me in the context of the whole workflow; from a staff member's perspective, for instance, I'd expect `has_document` to be `True` for all `WorkflowPhases` except `no_doc` and `failed`... Should this property be called `has_published_document` instead, maybe? Or am I misinterpreting things?

Reply:
You've got it right; I _suspect_ we'll be getting rid of this (and the "public" field) relatively soon, but I'm happy to update.